WcaOnRails/app/webpacker/components/CompetitionForm/CompsTable.js

```diff
@@ -0,0 +1,168 @@
+      <Table.HeaderCell name="date" width={2}>
+        {I18n.t('competitions.adjacent_competitions.date')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="location" width={2}>
+        {I18n.t('competitions.adjacent_competitions.location')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="distance" width={1}>
+        {I18n.t('competitions.adjacent_competitions.distance')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="limit" width={1}>
+        {I18n.t('competitions.adjacent_competitions.limit')}
+      </Table.HeaderCell>
+      <Table.HeaderCell name="competitors" width={1}>
+        {I18n.t('competitions.adjacent_competitions.competitors')}
+      </Table.HeaderCell>
+      {showEvents && (
+        <Table.HeaderCell name="events" width={2}>
+          {I18n.t('competitions.adjacent_competitions.events')}
+        </Table.HeaderCell>
+      )}
+    </Table.Row>
+  );
+}
+
+function CompsTableCompRow({ comp, action, showEvents }) {
+  return (
+    <Table.Row warning={!comp.danger} error={comp.danger}>
+      <Table.Cell name="name" width={3}>
+        <span dangerouslySetInnerHTML={{ __html: comp.nameLink }} />
+        {comp.confirmed && <NotConfirmedIcon />}
+        {action && (
+          <>
+            <br />
+            <Button
+              size="mini"
+              onClick={() => action.onClick(comp)}
+            >
+              {action.label}
+            </Button>
+          </>
+        )}
+      </Table.Cell>
+      <Table.Cell name="delegates" width={2}>
+        <span dangerouslySetInnerHTML={{ __html: comp.delegates }} />
+      </Table.Cell>
+      <Table.Cell name="date" width={2} data-toggle="tooltip" data-placement="top" data-container="body" title={comp.days_until_tooltip}>
+        {comp.days_distance_in_words}
+        {' '}
+        {comp.days_until < 0 ? I18n.t('competitions.adjacent_competitions.before') : I18n.t('competitions.adjacent_competitions.after')}
+      </Table.Cell>
+      <Table.Cell name="location" width={2}>
+        {comp.location}
+      </Table.Cell>
+      <Table.Cell name="distance" width={2} singleLine>
+        <span dangerouslySetInnerHTML={{ __html: comp.distance }} />
+      </Table.Cell>
+      <Table.Cell name="limit" width={1}>
+        {comp.limit ? comp.limit : ''}
+      </Table.Cell>
+      <Table.Cell name="competitors" width={1}>
+        {comp.competitors ? comp.competitors : ''}
+      </Table.Cell>
+      {showEvents && (
+        <Table.Cell name="events" width={2}>
+          {comp.events.map((e) => <span key={e} dangerouslySetInnerHTML={{ __html: e }} />)}
+        </Table.Cell>
+      )}
+    </Table.Row>
+  );
+}
+
+function CompsTableContent({ comps, action }) {
+  const [showEvents, setShowEvents] = useState(false);
+
+  return (
+    <Table structured>
+      <Table.Header>
+        <CompsTableHeaderRow showEvents={showEvents} />
+      </Table.Header>
+      <Table.Body>
+        {comps.reverse().map((comp) => (
```

Comment:
`reverse` reverses the array in place, which could lead to unintended consequences here or in the parent component. Probably best to make a copy? `[...comps].reverse().map(...)`